main.go

```diff
@@ -134,11 +135,16 @@ func cmd() *cobra.Command {
 		},
 	}
 
-	cmd.PersistentFlags().StringVar(&maxVersion, "provider-version", "",
-		`Upgrade the provider to the passed in version.
+	cmd.PersistentFlags().StringVar(&targetVersion, "target-version", "",
+		`Upgrade the provider to the passed version.
 
 If the passed version does not exist, an error is signaled.`)
 
+	cmd.PersistentFlags().BoolVar(&inferVersion, "pulumi-infer-version", false,
+		`Use our GH issues to infer the target upgrade version.
+		If both '--target-version' and '--pulumi-infer-version' are passed,
+		we take '--target-version' to cap the inferred version. [Hidden behind PULUMI_DEV]`)
```

Comment:
This should be actually hidden behind `PULUMI_DEV`. See `cmd.PersistentFlags().MarkHidden`.